src/Components/signUp/SignUp.js

```diff
@@ -0,0 +1,245 @@
+import { Fragment, useState, useEffect } from "react";
+import { Navigate } from "react-router-dom";
+
+import { Button, TextField, InputAdornment } from "@mui/material";
+import VisibilityIcon from "@mui/icons-material/Visibility";
+import { VisibilityOff } from "@mui/icons-material";
+
+import Image from "../image/ImageContainer";
+import { styles } from "../../utils";
+import "./styleSignUp.scss";
+
+const SignUp = () => {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [logInAccount, setLogInAccount] = useState(false);
+  const [showPassword, setShowPassword] = useState(false);
+  const [showRepeatPassword, setShowRepeatPassword] = useState(false);
+  const [nameUser, setNameUser] = useState("");
+  const [lastNameUser, setLastNameUser] = useState("");
+  const [emailUser, setEmailUser] = useState("");
+  const [passwordUser, setPasswordUser] = useState("");
+  const [repeatPasswordUser, setRepeatPasswordUser] = useState("");
+  const [messageSignUp, setMessageSignUp] = useState({
+    text: "Enter your detalls below.",
+    messageColor: { color: "rgb(150, 157, 166)" },
+  });
+  const userInformation = JSON.parse(localStorage.getItem("userInformation"));
+  const { classes } = styles();
+
+  useEffect(() => {
+    if (userInformation.isLoggedIn) {
+      setIsLoggedIn(true);
+    }
+  }, []);
+
+  const enterDashboardFunction = (e) => {
+    if (userInformation !== null && userInformation.email === emailUser) {
+      theUserExistsMessage();
+    } else if (passwordUser === repeatPasswordUser && passwordUser !== "") {
+      saveInformationUser();
+      setIsLoggedIn(true);
+    } else {
+      passWordNotMatchMessage();
+    }
+  };
+
+  const theUserExistsMessage = () => {
+    setMessageSignUp({
+      text: "The user already exists. Log In.",
+      messageColor: { color: "rgb(237, 85, 151)" },
+    });
+  };
+
+  const passWordNotMatchMessage = () => {
+    setMessageSignUp({
+      text: "Passwords do not match.",
+      messageColor: { color: "rgb(237, 85, 151)" },
+    });
+  };
+
+  const saveInformationUser = (name, lastName, email, password) => {
+    let informationUser = null;
+
+    informationUser = {
+      name: nameUser,
+      lastName: lastNameUser,
+      email: emailUser,
+      password: passwordUser,
+      isLoggedIn: true,
+    };
+
+    localStorage.setItem("userInformation", JSON.stringify(informationUser));
+  };
+
+  if (isLoggedIn) {
+    return <Navigate to="/dashboard" />;
+  } else if (logInAccount) {
```

Comment:
I think we can remove this conditional, let the user to come here when he wants.